### PR TITLE
Retire ZpDlog from crypto/dlogs

### PR DESCRIPTION
**Pull request by mancabizjak**
This PR includes refactoring that retires struct `ZpDlog`.

The file `zp.go` from `crypto/dlog`  package is removed, all its references are updated to make use of struct `SchnorrGroup` from `crypto/groups` package. Some variables and function names are renamed for clarity.

### CODE REVIEW

**Comment by mancabizjak** [comment]
@miha-stopar There, I implemented a `TODO` that was in the new `crypto/groups/schnorr.go`, which was supposed to replace `crypto/dlog/zp.go`. Please let me know if you prefer to rename specific function names (that still have "DLog" in their names) too.

File: crypto/zkp/primitives/dlogproofs/dlog_equality_blinded_transcript.go [hunk 1]
```diff
@@ -106,12 +106,12 @@ func (prover *DLogEqualityBTranscriptProver) GetProofData(challenge *big.Int) *b
...
 	z.Mul(challenge, prover.secret)
 	z.Add(z, prover.r)
-	z.Mod(z, prover.DLog.GetOrderOfSubgroup())
+	z.Mod(z, prover.Group.Q)
 	return z
 }
 
-type DLogEqualityBTranscriptVerifier struct {
-	DLog       *dlog.ZpDLog
+type GroupEqualityBTranscriptVerifier struct {
```
Comment (miha-stopar) [inline crypto/zkp/primitives/dlogproofs/dlog_equality_blinded_transcript.go]: DLogEqualityBTranscriptVerifier should stay named as it is - it uses group, but it is for verification of dlog equality (group is only a place where dlog "happens").

File: crypto/zkp/primitives/dlogproofs/dlog_equality_blinded_transcript.go [hunk 2]
```diff
@@ -124,20 +124,20 @@ type DLogEqualityBTranscriptVerifier struct {
 	transcript *Transcript
 }
 
-func NewDLogEqualityBTranscriptVerifier(dlog *dlog.ZpDLog,
-	gamma *big.Int) *DLogEqualityBTranscriptVerifier {
+func NewDLogEqualityBTranscriptVerifier(group *groups.SchnorrGroup,
+	gamma *big.Int) *GroupEqualityBTranscriptVerifier {
```
Comment (miha-stopar) [inline crypto/zkp/primitives/dlogproofs/dlog_equality_blinded_transcript.go]: DLogEqualityBTranscriptVerifier  should stay as it is.

File: crypto/zkp/primitives/dlogproofs/dlog_equality_blinded_transcript.go [hunk 3]
```diff
@@ -180,23 +180,23 @@ func (verifier *DLogEqualityBTranscriptVerifier) GetChallenge(g1, g2, t1, t2, x1
 
 // It receives z = r + secret * challenge.
 //It returns true if g1^z = g1^r * (g1^secret) ^ challenge and g2^z = g2^r * (g2^secret) ^ challenge.
-func (verifier *DLogEqualityBTranscriptVerifier) Verify(z *big.Int) (bool, *Transcript,
+func (verifier *GroupEqualityBTranscriptVerifier) Verify(z *big.Int) (bool, *Transcript,
```
Comment (miha-stopar) [inline crypto/zkp/primitives/dlogproofs/dlog_equality_blinded_transcript.go]: As above - DLogEqualityBTranscriptVerifier should stay named as it is.

File: crypto/zkp/primitives/dlogproofs/schnorr.go [hunk 4]
```diff
@@ -44,26 +44,26 @@ func ProveDLogKnowledge(secret, g1, t1 *big.Int, dlog *dlog.ZpDLog) bool {
...
+		Group:        group,
 		protocolType: protocolType,
 	}
 
 	if protocolType != types.Sigma {
 		// TODO: currently Pedersen is using the same dlog as SchnorrProver, this
 		// is because SchnorrVerifier for ZKP/ZKPOK needs to know Pedersen's dlog
 		// to generate a challenge and create a commitment
-		prover.PedersenReceiver = commitments.NewPedersenReceiverFromExistingDLog(dlog)
+		prover.PedersenReceiver = commitments.NewPedersenReceiverFromExistingGroup(group)
```
Comment (miha-stopar) [inline crypto/zkp/primitives/dlogproofs/schnorr.go]: As opposed to above - renaming into *Group here is fine.

**Comment by miha-stopar** [comment]
No, no need for renaming. Structs such as DLogEqualityBTranscriptVerifier and DLogEqualityProver should have DLog in its name, because they are about dlogs (as I said in a comment, group is only a place where dlog is computed).

**Pull request merged**